libraries/AP_Scripting/applets/rev_thr_on_land.lua

```diff
@@ -0,0 +1,151 @@
+    - FWD_THRUST_IDLE_PWM_ - min value of the motors when in forward throttle mode, set this to REV_THRUST_IDLE_PWM_ for smooth operation
+    - REV_THRUST_CHG_PWM_PS_(pwm) - this is the motors' linear acceleration time in seconds, from REV_THRUST_IDLE_PWM_ to REV_THRUST_HIGH_PWM_
+    - MAXTIME_REV_THRUST_APPLIED_MS_ (ms) - maximum time, in seconds, reverse throttle is allowed to apply
+    - TIME_THROT_WAIT_MS_ (ms) - time the throttle is suppressed, sitting at forward idle (used to ensure props are not spinning at a rate higher 
+      than FWD_THRUST_IDLE_PWM_ shortly before reverse throttle is applied)
+    - REV_THRUST_SPOOL_DPWM_ (pwm) - initial increase in pwm to apply when in reverse throttle mode
+]]--
+
+
+-- *****---PARAMS TO CHANGE---*****
+--REV arming parameters
+local RC_ARM_IN_CHANNEL_ = 6
+local ARM_CHANNEL_HI_PWM_ = 1500
+
+--REV switch setup
+local RC_REV_THROTTLE_CHANNEL_NUM_ = 11
+local REV_THRUST_SW_LOW_PWM_ = 1000
+local REV_THRUST_SW_HIGH_PWM_ = 2000
+
+--REV activation and spool up parameters
+local LAND_DET_TIME_ = 1
+local THRESH_LAND_HEIGHT_ = 0.12
+local TIME_THROT_WAIT_MS_ = 1000
+local REV_THRUST_SPOOL_DPWM_ = 60
+local REV_THRUST_CHG_PWM_PS_ = 150
+
+--REV strength and duration setup
+local FWD_THRUST_IDLE_PWM_ = 990
+local REV_THRUST_IDLE_PWM_ = 990
+local REV_THRUST_HIGH_PWM_ = 1600
+local MAXTIME_REV_THRUST_APPLIED_MS_ = 10000
+local MIN_GROUND_SPEED_ = -1
+
+-- *****---INTERNAL VARIABLES AND BOOL FLAGS --- DO NOT CHANGE ANYTHING HERE UNLESS MODIFYING THE CODE---*****
+local RFND_ROT_DN_ = 25
+local MAX_DIST_ = rangefinder:max_distance_cm_orient(RFND_ROT_DN_) * 0.01
+local MODE_FBWA = 5
+local COUNT_IN_RANGE_ = 0
+local THROT_CHANNEL_ = rc:get_channel(RCMAP_THROTTLE:get())
+local REV_THRUST_AUX_SW_ = rc:get_channel(RC_REV_THROTTLE_CHANNEL_NUM_)
+local TIME_SEC_ = 0
+local CALLBACK_TIME_ = 200
+local IS_THR_SUP_ = true
+local IS_IN_REV_THRUST_ = false
+local REV_THRUST_SW_ = false
+
+local REV_THRUST_CUR_PWM_ = FWD_THRUST_IDLE_PWM_ + REV_THRUST_SPOOL_DPWM_
+local TIM_AFTER_TOUCHDN_ = LAND_DET_TIME_ * 1000 * 0.01
+
+function UPDATE()
+  if (vehicle:get_mode() == MODE_FBWA and (arming:is_armed())
+      and (rc:get_pwm(RC_ARM_IN_CHANNEL_) > ARM_CHANNEL_HI_PWM_)) then
+    if (RFND_READING() < THRESH_LAND_HEIGHT_) then
+      if (COUNT_IN_RANGE_ > TIM_AFTER_TOUCHDN_) then
+        TIME_SEC_ = millis():tofloat()
+        return ENGAGE_REV_THRUST()
+      end
+      COUNT_IN_RANGE_ = COUNT_IN_RANGE_ + 1
+    else
+      COUNT_IN_RANGE_ = 0
+    end
+  end
+  return UPDATE, 100 -- check again in 10Hz
+end
+
+function RFND_READING()
+  if rangefinder:has_data_orient(RFND_ROT_DN_) then
+    return rangefinder:distance_cm_orient(RFND_ROT_DN_) * 0.01
+  end
+  return MAX_DIST_
+end
+
+function DISENGAGE_REV_THRUST()
+  THROT_CHANNEL_:set_override(REV_THRUST_IDLE_PWM_)
+  REV_THRUST_AUX_SW_:set_override(REV_THRUST_SW_LOW_PWM_)
```

Comment:
Should we override low? Or just lest the high override timeout? Currently this will override the pilot if they were to move the switch?

Reply:
The low is overriden to make sure we have idle forward thrust, to be consistent with the switch position that the pilot has had at and after the touchown.